src/goto-instrument/contracts/contracts.cpp

```
@@ -1445,67 +1451,43 @@ void code_contractst::add_contract_check(
   is_fresh_enforcet visitor(*this, log, wrapper_function);
   visitor.create_declarations();
   visitor.add_memory_map_decl(check);
+
   // Generate: assume(requires)
-  exprt::operandst requires_conjuncts;
-  for(const auto &r : code_type.requires())
-  {
-    requires_conjuncts.push_back(
-      to_lambda_expr(r).application(instantiation_values));
-  }
-  auto requires = conjunction(requires_conjuncts);
-  requires.add_source_location() =
-    requires_conjuncts.empty() ? code_type.source_location()
-                               : code_type.requires().front().source_location();
-  if(!requires.is_false())
+  for(const auto &clause : code_type.requires())
   {
-    if(has_subexpr(requires, ID_exists) || has_subexpr(requires, ID_forall))
-      add_quantified_variable(requires, function_symbol.mode);
+    auto instantiated_clause =
+      to_lambda_expr(clause).application(instantiation_values);
 
-    goto_programt assumption;
-    converter.goto_convert(
-      code_assumet(requires), assumption, function_symbol.mode);
-    visitor.update_requires(assumption);
-    check.destructive_append(assumption);
+    if(!instantiated_clause.is_false())
+    {
```

If the clause happens to be false maybe we should throw a hard error

---

Maybe yes. We should tell the user whether they are trying to assume a false assumption.

---

At this stage, code like
```c
__CPROVER_requires(false)
__CPROVER_requires(0)
```
is always represented as a `notequal` exprt, i.e., `0 != 0`. So this if statement is never reached. Regardless, I added this check in cause we improve/change our front-end in the future.